apps/wallet-mobile/src/features/SetupWallet/useCases/RestoreWallet/RestoreWalletScreen.tsx

```diff
@@ -110,32 +105,26 @@ export const RestoreWalletScreen = () => {
     }, [mnemonicType, track]),
   )
 
-  const handleOpenWalletWithDuplicatedName = React.useCallback(
-    (walletMeta: Wallet.Meta) => {
-      walletManager.setSelectedWalletId(walletMeta.id)
-      resetToTxHistory()
-    },
-    [walletManager, resetToTxHistory],
-  )
-
   const handleOnNext = React.useCallback(async () => {
-    const {csl, release} = wrappedCsl()
+    const {accountPubKeyHex} = await walletManager.generateWalletKeys(walletImplementation, mnemonic, accountVisual)
 
-    const {accountPubKeyHex} = await keyManager(walletImplementation)({mnemonic, csl, accountVisual})
-    const checksum = walletChecksum(accountPubKeyHex)
-    release()
+    const isWalletAccountDuplicated = walletManager.isWalletAccountDuplicated(accountPubKeyHex)
+
+    if (isWalletAccountDuplicated) {
+      const {plate, seed} = walletManager.checksum(accountPubKeyHex)
+      const duplicatedAccountWalletMeta = Array.from(walletManager.walletMetas.values()).find(
+        (walletMeta) => walletMeta.plate === plate,
+      )
 
-    const duplicatedWalletMeta = Array.from(walletManager.walletMetas.values()).find(
-      (walletMeta) => walletMeta.plate === checksum.TextPart,
-    )
+      if (duplicatedAccountWalletMeta === undefined) throw new Error('Restore Duplicated Wallet Error: invalid state')
```

Comment:
```suggestion
```

Reply:
We need to display the plate info in the dialog

Reply:
applied a refactor

https://github.com/Emurgo/yoroi/pull/3590/commits/252d95b14dff969694af190aed573eae341f477b